libbeat/publisher/queue/diskqueue/segments.go

```diff
@@ -470,6 +470,35 @@ func (segments *diskQueueSegments) sizeOnDisk() uint64 {
 	return total
 }
 
+// Iterates through all segment types to find the oldest ID in the queue
+func (segments *diskQueueSegments) oldestIDOnDisk() segmentID {
+	// not all segment types are pre-sorted, hence us appending some, taking [0] from others
+	oldSegments := []*queueSegment{}
+
+	if len(segments.reading) > 0 {
+		oldSegments = append(oldSegments, segments.reading[0])
+	}
+	if len(segments.acking) > 0 {
+		oldSegments = append(oldSegments, segments.acking[0])
+	}
+
+	oldSegments = append(oldSegments, segments.writing...)
+	oldSegments = append(oldSegments, segments.acked...)
+
+	sort.Slice(oldSegments, func(i, j int) bool { return oldSegments[i].id < oldSegments[j].id })
+
+	return oldSegments[0].id
+}
+
+// unACKedReadBytes returns the total count of bytes that have been read, but not ack'ed by the consumer
+func (segments *diskQueueSegments) unACKedReadBytes() uint64 {
+	var acc uint64
+	for _, seg := range segments.acking {
+		acc += seg.byteCount
```

Comment:
I'm not sure this is right either, because this will count all the bytes in a segment which may consist of all the events in the segment file. Ideally we would want this to increment event by event, otherwise it will just always be the size of the segment file currently being sent.

The disk queue is a collection of segment files, each segment file containing multiple frames (events): https://github.com/elastic/beats/blob/main/libbeat/publisher/queue/diskqueue/docs/on-disk-structures.md

Reply:
If this is difficult to implement in a useful way, I am in favour of just not implementing the metric at this point.

Reply:
@leehinman can you chime in here? I'm kind of going off the assumption that if we're reporting values in bytes, it doesn't matter too much how those byte segments map to individual events, but I don't know enough about the backend of the disk queue to know if we're just reporting the size of the currently sent segment, which might be less useful?

Reply:
I don't think the amount of events (or bytes) that have been read but not acked would be very useful.  It should always be the same as amount of data the output has sent to but hasn't heard back yet.  So the  output should have this info.